[Robotics] Humanoid/ethercat

intelzhangw2 opened the pull request:
### Description

Add robotmctask to integrate RL pipeline into motion control task for humanoid. This features refer to ethercat stack, ecat-enablekit and new robotmctask. Also add ti5_rviz package as new ros2 node into motion-control-gateway

Fixes # (issue)

### Any Newly Introduced Dependencies

Please describe any newly introduced 3rd party dependencies in this change. List their name, license information and how they are used in the project.

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### Checklist:

- [x] I agree to use the APACHE-2.0 license for my code changes.
- [x] I have not introduced any 3rd party components incompatible with APACHE-2.0. 
- [x] I have not included any company confidential information, trade secret, password or security token. 
- [x] I have performed a self-review of my code.

intelzhangw2 committed:
update ethercat stack to add new api and ecrt configuration

Signed-off-by: Zhang, Wei E <wei.e.zhang@intel.com>

intelzhangw2 committed:
add new api/examples for multiple masters and bugfix

Signed-off-by: Zhang, Wei E <wei.e.zhang@intel.com>

intelzhangw2 committed:
import new Robotmctask component

Signed-off-by: Zhang, Wei E <wei.e.zhang@intel.com>

intelzhangw2 committed:
update license header for robotmctask

Signed-off-by: Zhang, Wei E <wei.e.zhang@intel.com>

intelzhangw2 committed:
Add robot_rviz package for ti5 humanoid robot

Signed-off-by: Zhang, Wei E <wei.e.zhang@intel.com>

jb-balaji reviewed (changes requested):
Fix memory leak issues

jb-balaji committed:
Fix issues found in PR1305 review

->Fix thread management in dual_master_motor.c:
* Add separate thread variable for second thread * Fix null pointer check for master1
* Wait for both threads with pthread_join * Release both masters in cleanup paths

-> Fix memory leaks in motionentry.c:
  * Clean up first master in error path of motion_servo_master_create_v2
  * Fix incorrect NULL returns (should be ECAT_FAIL) in motion_servo_slave_config_register_v2
  * Add proper cleanup for domain allocations on error

-> Improve memory safety in mc_rl_sample.cpp:
  * Add null checks after malloc
  * Set pointer to nullptr after free
  * Add config_file cleanup at program exit"

Signed-off-by: Jayabalaji <jayabalaji.sathiyamoorthi@intel.com>

intelzhangw2 committed:
update ethercat stack patch to improve on new API

Signed-off-by: Zhang, Wei E <wei.e.zhang@intel.com>

jb-balaji reviewed (approved):
Approved.

jouillet commented on libraries/edge-control-libraries/fieldbus/robotmctask/README.md:
````diff
@@ -0,0 +1,72 @@
+2. install plcopen-motion library
+
+```shell
+sudo apt install plcopen-motion-dev plcopen-servo-dev   plcopen-ruckig-dev plcopen-databus-dev plcopen-benchmark-dev
````
Need to explain to user how to onboard the ECI APT repo prior to running these install commands.  Can probably link to the ECI documentation.
